code/modules/mob/living/carbon/human/species/station/skrell/skrell.dm

```diff
@@ -61,8 +61,39 @@
 
 	zombie_type = "Skrell Zombie"
 
+/datum/species/skrell/handle_environment_special(var/mob/living/carbon/human/H)
+	var/old_moisture = H.ST.moisture
+	H.ST.moisture = max(MOISTURE_NONE, H.ST.moisture - rand(8, 13)/10)			// Reduce moisture a little.
+	if(H.fire_stacks && H.ST.moisture > MOISTURE_NORMAL)			// Flaming moist skrell can reduce their flame
+		H.fire_stacks--
+		H.ST.moisture -= rand(150, 350)								// Lost a lot of moisture
+	var/pain_limit = (MOISTURE_NORMAL-H.ST.moisture) * (40 / MOISTURE_NORMAL)	// The max halloss damage to do based on current moisture. At 0, this is 30. At 1800 or higher, this is 0.
+	
+	// Add more pain damage if they need it.
+	var/setPain = max(H.getHalLoss(), pain_limit)
+	H.adjustHalLoss(setPain - H.getHalLoss())
+	
+	// Skrell hydration is directly tied to their moisture.
+	H.hydration = (H.ST.moisture / MOISTURE_MAX) * H.max_hydration
+
+	if(H.ST.moisture <= MOISTURE_NORMAL && old_moisture > MOISTURE_NORMAL)
+		to_chat(H, SPAN_WARNING("You are drying out. You should consider moisturizing."))
+	else if(H.ST.moisture <= MOISTURE_LOW && old_moisture > MOISTURE_LOW)
+		to_chat(H, SPAN_WARNING("You are really dry! Your skin feels uncomfortable and flakey!"))
+	else if(H.ST.moisture == MOISTURE_NONE && old_moisture > MOISTURE_NONE)
+		to_chat(H, SPAN_DANGER("You are completely dry, this is really painful!"))
```

Comment:
This will be VERY spammy.

Reply:
Can only happen once per level. Should only happen once every ~5-10 minutes